TheMagicApprentice/modules/entities/unicorns/Unicorn.cs

```diff
@@ -0,0 +1,84 @@
+using Godot;
+using System;
+
+public partial class Unicorn : CharacterBody2D
+{
+	[Export]
+	public StateMachine StateMachine; ///< Reference to the state machine of the unicorn
+	[Export]
+	public AnimationPlayer AnimationPlayer; ///< Reference to the animation player of the unicorn
+
+	[Export]
+	public float MeleeAttackRange = 50; ///< If the PC is inside of this radius, the unicorn performs a melee attack
+
+	private MagicType _magicType; ///< Magic type of the unicorn
+
+	/**
+	Is called when the unicorn enters the scene tree.
+	Checks if the references to the state machine and the animation player are valid and then sends them to the state machine so that all states get the references
+	*/
+	public override void _Ready()
+	{
+		System.Diagnostics.Debug.Assert(StateMachine is not null);
```

Comment:
When using Debug.Assert always remember to give an error string. Otherwise it is impossible to know where the problem was